core/src/com/unciv/logic/map/MapUnit.kt

```diff
@@ -111,11 +111,16 @@ class MapUnit {
     fun displayName(): String {
         val baseName =
             if (instanceName == null) "[$name]"
-            else "instanceName ([$name])"
+            else "$instanceName ([$name])"
 
         return if (religion == null) baseName
         else "$baseName ([${getReligionDisplayName()}])"
     }
+    
+    fun shortDisplayName(): String {
+        return if (instanceName != null) "[$instanceName]"!!
```

Comment:
This is definitely a `UNNECESSARY_NOT_NULL_ASSERTION` - a string literal, even with templates, is never null. Any template placeholders evaluating to null will be replaced with the letters "null", so ensuring non-null on them is always optional, too. Just FYI.

Reply:
I have no idea how that even ended up there, I fully agree that it looks completely superfluous